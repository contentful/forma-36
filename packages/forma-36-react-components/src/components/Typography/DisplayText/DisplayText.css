@import 'settings/colors';
@import 'settings/typography';
@import 'settings/transitions';

.DisplayText {
  display: block;
  font-family: var(--font-stack-primary);
  font-weight: var(--font-weight-demi-bold);
  color: var(--color-text-dark);
  text-rendering: optimizeLegibility;
  margin: 0;
}

.DisplayText--default {
  font-size: var(--font-size-2xl);
  line-height: var(--line-height-2xl);
}

.DisplayText--large {
  font-size: var(--font-size-3xl);
  line-height: var(--line-height-3xl);
}

.DisplayText--huge {
  font-size: var(--font-size-4xl);
  line-height: var(--line-height-4xl);
}
