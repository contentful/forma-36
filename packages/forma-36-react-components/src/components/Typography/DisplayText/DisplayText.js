import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './DisplayText.css';

class DisplayText extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node.isRequired,
    testId: PropTypes.string,
    element: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p']),
    size: PropTypes.oneOf(['default', 'large']),
  };

  static defaultProps = {
    element: 'h1',
    extraClassNames: undefined,
    testId: 'cf-ui-display-text',
    size: 'default',
  };

  render() {
    const {
      extraClassNames,
      children,
      testId,
      element,
      size,
      ...otherProps
    } = this.props;

    const classNames = cn(styles.DisplayText, extraClassNames, {
      [styles[`DisplayText--${size}`]]: size,
    });

    const Element = element;

    return (
      <Element className={classNames} data-test-id={testId} {...otherProps}>
        {children}
      </Element>
    );
  }
}

export default DisplayText;
