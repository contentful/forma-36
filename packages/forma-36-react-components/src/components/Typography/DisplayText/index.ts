export * from './DisplayText';
export { default } from './DisplayText';
