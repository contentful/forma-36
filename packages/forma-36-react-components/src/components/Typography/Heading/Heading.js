import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './Heading.css';

class Heading extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node.isRequired,
    testId: PropTypes.string,
    element: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p']),
  };

  static defaultProps = {
    element: 'h1',
    extraClassNames: undefined,
    testId: 'cf-ui-heading',
  };

  render() {
    const {
      extraClassNames,
      children,
      testId,
      element,
      ...otherProps
    } = this.props;

    const classNames = cn(styles.Heading, extraClassNames);

    const Element = element;

    return (
      <Element className={classNames} data-test-id={testId} {...otherProps}>
        {children}
      </Element>
    );
  }
}

export default Heading;
