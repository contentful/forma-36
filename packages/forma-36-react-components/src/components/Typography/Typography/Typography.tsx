import React, { Component } from 'react';
import cn from 'classnames';

import styles from './Typography.css';

export interface TypographyProps {
  className?: string;
  children?: React.ReactNode;
  style?: React.CSSProperties;
  testId?: string;
}

const defaultConfiguration = {
  displayText: { spacing: 'l' },
  displayTextLarge: { spacing: 'xl' },
  heading: { spacing: 'm' },
  paragraph: { spacing: 'm' },
  sectionHeading: { spacing: 'l' },
  subheading: { spacing: 'm' },
};

const defaultProps: Partial<TypographyProps> = {
  testId: 'cf-ui-text-container',
};

export const TypographyContext = React.createContext({});

export class Typography extends Component<TypographyProps> {
  static defaultProps = defaultProps;

  render() {
    const { className, children, testId, ...otherProps } = this.props;
    const classNames = cn(styles.Typography, className);

    return (
      <TypographyContext.Provider value={defaultConfiguration}>
        <div {...otherProps} className={classNames} data-test-id={testId}>
          {children}
        </div>
      </TypographyContext.Provider>
    );
  }
}

export default Typography;
