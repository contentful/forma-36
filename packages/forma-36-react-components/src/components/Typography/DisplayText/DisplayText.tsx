import React from 'react';
import cn from 'classnames';
import styles from './DisplayText.css';

import { TypographyContext } from '../Typography';

export interface DisplayTextProps {
  element?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';
  size?: 'default' | 'large' | 'huge';
  className?: string;
  children?: React.ReactNode;
  style?: React.CSSProperties;
  testId?: string;
}

export function DisplayText({
  className,
  children,
  testId = 'cf-ui-display-text',
  element = 'h1',
  size = 'default',
  ...otherProps
}: DisplayTextProps): React.ReactElement {
  const classNames = cn(styles['DisplayText'], className, {
    [styles[`DisplayText--${size}`]]: size,
  });

  const Element = element!; // eslint-disable-line @typescript-eslint/no-non-null-assertion

  return (
    <TypographyContext.Consumer>
      {(value) => {
        const contextSize =
          size === 'large' ? value['displayTextLarge'] : value['displayText'];

        return (
          <Element
            className={cn(classNames, [
              contextSize && `f36-margin-bottom--${contextSize.spacing}`,
            ])}
            data-test-id={testId}
            {...otherProps}
          >
            {children}
          </Element>
        );
      }}
    </TypographyContext.Consumer>
  );
}

export default DisplayText;
