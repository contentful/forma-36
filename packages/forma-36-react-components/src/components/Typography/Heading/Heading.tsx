import React, { Component } from 'react';
import cn from 'classnames';
import styles from './Heading.css';

import { TypographyContext } from '../Typography/Typography';

export interface HeadingProps {
  extraClassNames?: string;
  children?: React.ReactNode;
  testId?: string;
  element: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';
  style?: React.CSSProperties;
}

export class Heading extends Component<HeadingProps> {
  static defaultProps = {
    element: 'h1',
    testId: 'cf-ui-heading',
  };

  render() {
    const {
      extraClassNames,
      children,
      testId,
      element,
      ...otherProps
    } = this.props;

    const classNames = cn(styles['Heading'], extraClassNames);

    const Element = element;

    return (
      <TypographyContext.Consumer>
        {value => {
          return (
            <Element
              className={cn(classNames, [
                value['heading'] &&
                  `f36-margin-bottom--${value['heading']['spacing']}`,
              ])}
              data-test-id={testId}
              {...otherProps}
            >
              {children}
            </Element>
          );
        }}
      </TypographyContext.Consumer>
    );
  }
}

export default Heading;
