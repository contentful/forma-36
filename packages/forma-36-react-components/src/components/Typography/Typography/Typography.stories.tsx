import React from 'react';

import Typography, { TypographyProps } from './Typography';
import DisplayText from '../DisplayText';
import Heading from './../Heading';
import Subheading from './../Subheading';
import SectionHeading from '../SectionHeading';
import Paragraph from './../Paragraph';
import Flex from '../../Flex/Flex';

export default {
  title: 'Typography/Typography',
  component: Typography,
  parameters: {
    propTypes: [Typography['__docgenInfo']],
  },
  argTypes: {
    className: { control: { disable: true } },
    testId: { control: { disable: true } },
  },
};

const loremIpsum = `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Facilis
accusantium vel voluptate incidunt, tempora consectetur consequuntur
magnam reiciendis ea ipsam!`;

interface Arg extends TypographyProps {
  title?: string;
  paragraph?: string;
}

export const WithDisplayHuge = ({ title, paragraph, ...args }: Arg) => {
  return (
    <ExampleContainer>
      <Typography {...args}>
        <DisplayText size="huge">{title}</DisplayText>
        <Paragraph>{paragraph}</Paragraph>
        <Paragraph>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Non ut
          accusamus quia debitis expedita consectetur!
        </Paragraph>
      </Typography>
    </ExampleContainer>
  );
};
WithDisplayHuge.args = {
  title: 'My DisplayText',
  paragraph: loremIpsum,
};

export const WithDisplayLarge = ({ title, paragraph, ...args }: Arg) => {
  return (
    <ExampleContainer>
      <Typography {...args}>
        <DisplayText size="large">{title}</DisplayText>
        <Paragraph>{paragraph}</Paragraph>
        <Paragraph>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Non ut
          accusamus quia debitis expedita consectetur!
        </Paragraph>
      </Typography>
    </ExampleContainer>
  );
};
WithDisplayLarge.args = {
  title: 'My DisplayText',
  paragraph: loremIpsum,
};

export const WithDisplayDefault = ({ title, paragraph, ...args }: Arg) => {
  return (
    <ExampleContainer>
      <Typography {...args}>
        <DisplayText>{title}</DisplayText>
        <Paragraph>{paragraph}</Paragraph>
        <Paragraph>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Non ut
          accusamus quia debitis expedita consectetur!
        </Paragraph>
      </Typography>
    </ExampleContainer>
  );
};
WithDisplayDefault.args = {
  title: 'My DisplayText',
  paragraph: loremIpsum,
};

export const WithHeading = ({ title, paragraph, ...args }: Arg) => {
  return (
    <ExampleContainer>
      <Typography {...args}>
        <Heading>{title}</Heading>
        <Paragraph>{paragraph}</Paragraph>
        <Paragraph>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Non ut
          accusamus quia debitis expedita consectetur!
        </Paragraph>
      </Typography>
    </ExampleContainer>
  );
};
WithHeading.args = {
  title: 'My Heading',
  paragraph: loremIpsum,
};

export const WithSubheading = ({ title, paragraph, ...args }: Arg) => {
  return (
    <ExampleContainer>
      <Typography {...args}>
        <Subheading>{title}</Subheading>
        <Paragraph>{paragraph}</Paragraph>
        <Paragraph>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Non ut
          accusamus quia debitis expedita consectetur!
        </Paragraph>
      </Typography>
    </ExampleContainer>
  );
};
WithSubheading.args = {
  title: 'My Subheading',
  paragraph: loremIpsum,
};

export const WithSectionHeading = ({ title, paragraph, ...args }: Arg) => {
  return (
    <ExampleContainer>
      <Typography {...args}>
        <SectionHeading>{title}</SectionHeading>
        <Paragraph>{paragraph}</Paragraph>
        <Paragraph>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Non ut
          accusamus quia debitis expedita consectetur!
        </Paragraph>
      </Typography>
    </ExampleContainer>
  );
};
WithSectionHeading.args = {
  title: 'My SectionHeading',
  paragraph: loremIpsum,
};

export const Overview = () => {
  return (
    <>
      <Flex alignItems="center" marginBottom="spacingM">
        <Flex marginRight="spacingS">
          <Paragraph>48</Paragraph>
        </Flex>
        <DisplayText size="huge">Display Huge</DisplayText>
      </Flex>
      <Flex alignItems="center" marginBottom="spacingM">
        <Flex marginRight="spacingS">
          <Paragraph>36</Paragraph>
        </Flex>
        <DisplayText size="large">Display Large</DisplayText>
      </Flex>
      <Flex alignItems="center" marginBottom="spacingM">
        <Flex marginRight="spacingS">
          <Paragraph>28</Paragraph>
        </Flex>
        <DisplayText size="default">Display</DisplayText>
      </Flex>

      <Flex alignItems="center" marginBottom="spacingM">
        <Flex marginRight="spacingS">
          <Paragraph>20</Paragraph>
        </Flex>
        <Heading>Heading</Heading>
      </Flex>

      <Flex alignItems="center" marginBottom="spacingM">
        <Flex marginRight="spacingS">
          <Paragraph>16</Paragraph>
        </Flex>
        <Subheading>Subheading</Subheading>
      </Flex>

      <Flex alignItems="center" marginBottom="spacingM">
        <Flex marginRight="spacingS">
          <Paragraph>12</Paragraph>
        </Flex>
        <SectionHeading>Section Heading</SectionHeading>
      </Flex>

      <Flex alignItems="center" marginBottom="spacingM">
        <Flex marginRight="spacingS">
          <Paragraph>14</Paragraph>
        </Flex>
        <Paragraph>Paragraph</Paragraph>
      </Flex>
    </>
  );
};

function ExampleContainer({ children }) {
  return <Flex style={{ maxWidth: '600px' }}>{children}</Flex>;
}
