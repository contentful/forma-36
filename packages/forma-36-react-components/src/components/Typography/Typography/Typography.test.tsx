import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { axe } from 'jest-axe';
import Typography from './Typography';
import DisplayText from './../DisplayText';
import Heading from './../Heading';
import Paragraph from './../Paragraph';
import SectionHeading from './../SectionHeading';
import Subheading from './../Subheading';

it('renders the component', () => {
  const output = shallow(
    <Typography>
      <Heading>My Heading</Heading>
      <Paragraph>My Paragraph</Paragraph>
    </Typography>,
  );

  expect(output).toMatchSnapshot();
});

it('renders all typography components with the expected spacing classes', () => {
  const output = render(
    <Typography>
      <DisplayText>My DisplayText</DisplayText>
      <Heading>My Heading</Heading>
      <Paragraph>My Paragraph</Paragraph>
      <SectionHeading>My SectionHeading</SectionHeading>
      <Subheading>My Subheading</Subheading>
    </Typography>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const output = shallow(
    <Typography className="my-extra-class">
      <Heading>My Heading</Heading>
      <Paragraph>My Paragraph</Paragraph>
    </Typography>,
  );

  expect(output).toMatchSnapshot();
});

it('has no a11y issues', async () => {
  const output = mount(
    <Typography>
      <Heading>My Heading</Heading>
      <Paragraph>My Paragraph</Paragraph>
    </Typography>,
  ).html();
  const results = await axe(output);

  expect(results).toHaveNoViolations();
});
