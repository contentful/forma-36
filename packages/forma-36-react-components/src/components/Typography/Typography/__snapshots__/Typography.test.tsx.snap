// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all typography components with the expected spacing classes 1`] = `
.emotion-0 {
  display: block;
  margin-bottom: 1rem;
  display: block;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-weight: 600;
  color: #192532;
  font-size: 1.25rem;
  line-height: 2rem;
  text-rendering: optimizeLegibility;
  margin: 0;
}

<div
  class="Typography"
  data-test-id="cf-ui-text-container"
>
  <h1
    class="DisplayText DisplayText--default f36-margin-bottom--l"
    data-test-id="cf-ui-display-text"
  >
    My DisplayText
  </h1>
  <div
    class="emotion-0"
  >
    My Heading
  </div>
  <p
    class="Paragraph f36-margin-bottom--m"
    data-test-id="cf-ui-paragraph"
  >
    My Paragraph
  </p>
  <h3
    class="SectionHeading f36-margin-bottom--l"
    data-test-id="cf-ui-section-heading"
  >
    My SectionHeading
  </h3>
  <h2
    class="Subheading f36-margin-bottom--m"
    data-test-id="cf-ui-subheading"
  >
    My Subheading
  </h2>
</div>
`;

exports[`renders the component 1`] = `
.emotion-0 {
  display: block;
  margin-bottom: 1rem;
  display: block;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-weight: 600;
  color: #192532;
  font-size: 1.25rem;
  line-height: 2rem;
  text-rendering: optimizeLegibility;
  margin: 0;
}

<div
  class="Typography"
  data-test-id="cf-ui-text-container"
>
  <div
    class="emotion-0"
  >
    My Heading
  </div>
  <p
    class="Paragraph f36-margin-bottom--m"
    data-test-id="cf-ui-paragraph"
  >
    My Paragraph
  </p>
</div>
`;

exports[`renders the component with an additional class name 1`] = `
.emotion-0 {
  display: block;
  margin-bottom: 1rem;
  display: block;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-weight: 600;
  color: #192532;
  font-size: 1.25rem;
  line-height: 2rem;
  text-rendering: optimizeLegibility;
  margin: 0;
}

<div
  class="Typography my-extra-class"
  data-test-id="cf-ui-text-container"
>
  <div
    class="emotion-0"
  >
    My Heading
  </div>
  <p
    class="Paragraph f36-margin-bottom--m"
    data-test-id="cf-ui-paragraph"
  >
    My Paragraph
  </p>
</div>
`;
